testament/testament.nim

```diff
@@ -30,10 +30,32 @@ var backendLogging = true
 var simulate = false
 var optVerbose = false
 var useMegatest = true
+var optFailing = false
+
+## Blanket method to encaptsulate all echos while testament is detangled.
+## Using this means echo cannot be called with separation of args and must instead
+## pass a single concatenated string so that optional paramters can be
+## included
+type
+  MessageType = enum
+    Undefined,
+    Progress,
+    ProcessCmdCall
+
+proc msg(msgType: MessageType; parts: varargs[string, `$`]) =
+  block:
+    if optFailing and not optVerbose:
+      case msgType
+      of ProcessCmdCall:
+        break
```

Comment:
Can't the whole block break thing go away so a single if?

Reply:
It can; but thats not as extensible. I'll reduce it for now.